packages/labs/cli/src/lib/init/element-starter/templates/lib/element.ts

```diff
@@ -0,0 +1,164 @@
+import {FileTree} from '@lit-labs/gen-utils/lib/file-utils.js';
+import {
+  javascript,
+  kabobToPascalCase,
+} from '@lit-labs/gen-utils/lib/str-utils.js';
+import {Language} from '../../../../commands/init.js';
+
+export const generateElement = (
+  elementName: string,
+  language: Language
+): FileTree => {
+  const directory = language === 'js' ? 'lib' : 'src';
+  return {
+    [directory]: {
+      [`${elementName}.${language}`]:
+        language === 'js' ? js(elementName) : ts(elementName),
+    },
+  };
+};
+
+const js = (elementName: string) => {
+  const className = kabobToPascalCase(elementName);
+  return javascript`/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import { LitElement, html, css } from 'lit';
+
+/**
+ * An example element.
+ *
+ * @fires count-changed - Indicates when the count changes
+ * @slot - This element has a slot
+ * @csspart button - The button
+ */
+export class ${className} extends LitElement {
+  static get styles() {
+    return css\`
+      :host {
+        display: block;
+        border: solid 1px gray;
+        padding: 16px;
+        max-width: 800px;
+      }
+    \`;
+  }
+
+  static get properties() {
+    return {
+      /**
+       * The name to say "Hello" to.
+       * @type {string}
+       */
+      name: {type: String},
+
+      /**
+       * The number of times the button has been clicked.
+       * @type {number}
+       */
+      count: {type: Number},
+    };
+  }
+
+  constructor() {
+    super();
+    this.name = 'World';
+    this.count = 0;
+  }
+
+  render() {
+    return html\`
+      <h1>\${this._sayHello()}!</h1>
+      <button @click=\${this._onClick} part="button">
+        Click Count: \${this.count}
+      </button>
+      <slot></slot>
+    \`;
+  }
+
+  _onClick() {
+    this.count++;
+    this.dispatchEvent(new CustomEvent('count-changed'));
```

Comment:
I like making this `bubble` and `composed`.

Reply:
Consider putting this method before render to make it easer to delete "count" entirely?

Reply:
2 things:

1. If we make this composed, then you'd need to carry data with it – it may travel through shadow roots, so `event.target` may point to the wrong thing due to event retargetting
2. I don't think we should move count, because we have to balance ease of use for reusing this and writing readable code. And personally I think you should be organizing your methods by chronological order. e.g.

```ts
class MyEl extends LitElement {
  static properties;
  constructor()
  connectedCallback()
  update()
  willUpdate()
  render()
  renderSubtemplateCalledByRender()
  renderSubTemplateCalledByPreviousMethod()
  userInteractionCallbacksLikeClick()
  firstUpdated()
  updated()
  disconnectedCallback()
}
```

Reply:
Changed to normal Event and added bubbles but not composed for the reasons above